tiledb/api/c_api/array/array_api.cc

```diff
@@ -663,6 +663,13 @@ capi_return_t tiledb_array_load_all_enumerations(const tiledb_array_t* array) {
   return TILEDB_OK;
 }
 
+capi_return_t tiledb_array_load_enumerations_all_schemas(
```

Comment:
Adding this API instead of [changing parameters of tiledb_array_load_all_enumerations](https://github.com/TileDB-Inc/TileDB/pull/5349/commits/55a7168a4c0b72fa9f38aaf7e0c56da04d0cbe2b#diff-38be676bdc557702f3f0aea10fab4cc5a50d26419e898846321f07f0121e3c46L660-L661) allows this PR to fix the regression without changes to REST or other repositories. We will have a new binding to wrap but that's just for completeness, this will only be used by the client to submit the request to REST. Had we changed the parameters like I did initially REST would [fail to link](https://gitlab.com/TileDB-Inc/TileDB-Internal/-/jobs/8066740726) against this branch until we updated the modified binding for the APIs.

Reply:
I mean, we can't really change a C API after we have released, can we? Or that's some exception for experimentals? What we can do if it's incorrect though is mark it for deprecation. But rather than incorrect, is `tiledb_array_load_all_enumerations`  just fetching the enumerations for the latest schema only?
Then IMO it's good choice to add another API anyway since it's serving a different purpose and has different performance impact.

Reply:
IIUC for experimental APIs we can change them since they are experimental, but it's a breaking change and I was trying to avoid doing that. Since `tiledb_array_load_all_enumerations` only loads for the latest schema I thought a new API would not be out of the question.

Yeah I think the name `tiledb_array_load_all_enumerations` is a _little_ misleading as you probably noticed but I updated those docs to clarify and if we want to rename or restructure those methods I think we could do that separately and go through a deprecation cycle.